packages/core/src/notifications/activation.ts

```diff
@@ -24,25 +25,26 @@ const emergencyPollTime = oneMinute * 10
  * @param initialState initial auth state
  * @param authStateFn fn to get current auth state
  */
-export async function activate(
-    context: vscode.ExtensionContext,
-    initialState: AuthState,
-    authStateFn: () => Promise<AuthState>
-) {
+export async function activate(context: vscode.ExtensionContext, authStateFn: () => Promise<AuthState>) {
     try {
         const panelNode = NotificationsNode.instance
         panelNode.registerView(context)
 
-        const controller = new NotificationsController(panelNode)
-        const engine = new RuleEngine(await getRuleContext(context, initialState))
-
-        await controller.pollForStartUp(engine)
-        await controller.pollForEmergencies(engine)
-
-        globals.clock.setInterval(async () => {
-            const ruleContext = await getRuleContext(context, await authStateFn())
-            await controller.pollForEmergencies(new RuleEngine(ruleContext))
-        }, emergencyPollTime)
+        const controller = new NotificationsController(
+            panelNode,
+            async () => await getRuleContext(context, await authStateFn()),
+            DevSettings.instance.isDevMode() ? new DevFetcher() : new RemoteFetcher()
+        )
+
+        await controller.pollForStartUp()
```

Comment:
Are these making a network call? If so I wonder if we should not block to improve startup performance

Reply:
Yes excellent catch, this was a bug fixed by https://github.com/aws/aws-toolkit-vscode/pull/6052